src/MCPClient/lib/clientScripts/create_transfer_mets.py

```diff
@@ -0,0 +1,351 @@
+            to_attr="related_is_source_of",
+        ),
+    ]
+    try:
+        file_obj = File.objects.prefetch_related(*prefetches).get(**db_lookup)
+    except File.DoesNotExist:
+        logger.info("No record in database for file: %s", db_file_path)
+        return None
+
+    return file_obj
+
+
+def get_premis_format_data(file_ids):
+    format_data = ()
+
+    for file_id in file_ids:
+        format_data += (
+            "format",
+            (
+                "format_designation",
+                ("format_name", file_id.format_name),
+                ("format_version", file_id.format_version),
+            ),
+            (
+                "format_registry",
+                ("format_registry_name", file_id.format_registry_name),
+                ("format_registry_key", file_id.format_registry_key),
+            ),
+        )
+    if not format_data:
+        # default to unknown
+        format_data = ("format", ("format_designation", ("format_name", "Unknown")))
+
+    return format_data
+
+
+def get_premis_relationship_data(derivations, originals):
+    relationship_data = ()
+
+    for derivation in derivations:
+        relationship_data += (
+            (
+                "relationship",
+                ("relationship_type", "derivation"),
+                ("relationship_sub_type", "is source of"),
+                (
+                    "related_object_identification",
+                    ("related_object_identifier_type", "UUID"),
+                    ("related_object_identifier_value", derivation.derived_file_id),
+                ),
+                (
+                    "related_event_identification",
+                    ("related_event_identifier_type", "UUID"),
+                    ("related_event_identifier_value", derivation.event_id),
+                ),
+            ),
+        )
+    for original in originals:
+        relationship_data += (
+            (
+                "relationship",
+                ("relationship_type", "derivation"),
+                ("relationship_sub_type", "has source"),
+                (
+                    "related_object_identification",
+                    ("related_object_identifier_type", "UUID"),
+                    ("related_object_identifier_value", original.source_file_id),
+                ),
+                (
+                    "related_event_identification",
+                    ("related_event_identifier_type", "UUID"),
+                    ("related_event_identifier_value", original.event_id),
+                ),
+            ),
+        )
+
+    return relationship_data
+
+
+def file_obj_to_premis(file_obj):
+    """
+    Converts an File model object to a PREMIS event object via metsrw.
+
+    Returns:
+        metsrw.plugins.premisrw.premis.PREMISObject
+    """
+
+    premis_digest_algorithm = convert_to_premis_hash_function(file_obj.checksumtype)
+    premis_data = (
+        "object",
+        metsrw.plugins.premisrw.PREMIS_2_2_META,
```

Comment:
```suggestion
        metsrw.plugins.premisrw.PREMIS_3_0_META,
```

Reply:
Thanks! I was looking at the current behaviour of createprocessedstructmap. Will change.